profile/src/main/java/com/scottlogic/deg/profile/validators/profile/constraints/atomic/IntegerConstraintValidator.java

```diff
@@ -0,0 +1,49 @@
+package com.scottlogic.deg.profile.validators.profile.constraints.atomic;
+
+import com.scottlogic.deg.common.profile.FieldType;
+import com.scottlogic.deg.common.util.Defaults;
+import com.scottlogic.deg.common.validators.ValidationResult;
+import com.scottlogic.deg.profile.dtos.FieldDTO;
+import com.scottlogic.deg.profile.dtos.constraints.atomic.numeric.IntegerConstraintDTO;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+public class IntegerConstraintValidator extends AtomicConstraintValidator<IntegerConstraintDTO>
+{
+    private final FieldType expectedFieldType;
+
+    public IntegerConstraintValidator(String rule, List<FieldDTO> fields, FieldType expectedFieldType)
+    {
+        super(rule, fields);
+        this.expectedFieldType = expectedFieldType;
+    }
+
+    @Override
+    public final ValidationResult validate(IntegerConstraintDTO dto)
+    {
+        ValidationResult fieldMustBeValid = fieldMustBeValid(dto);
+        if(!fieldMustBeValid.isSuccess) return fieldMustBeValid;
+
+        ValidationResult integerMustBeValid = integerMustBeValid(dto);
+        if(!integerMustBeValid.isSuccess) return integerMustBeValid;
+
+        return fieldTypeMustMatchValueType(dto, expectedFieldType);
+    }
+
+    private ValidationResult integerMustBeValid(IntegerConstraintDTO dto)
+    {
+        BigDecimal integer = BigDecimal.valueOf(dto.getInt());
+        BigDecimal max = BigDecimal.valueOf(Defaults.MAX_STRING_LENGTH);
+        BigDecimal min = BigDecimal.ZERO;
+        if (integer.compareTo(min) < 0)
+        {
+            return ValidationResult.failure(String.format("String length must have a value >= %s, currently is %s", min.toPlainString(), integer.toPlainString()));
```

Comment:
Given this is a validator for string length, perhaps the name shouldn't be be `IntegerConstraintValidator`. When I was reading this I got confused as to why we are validating `0 < x < MAX_STRING_LENGTH`, when the signature is `integerMustBeValid(IntegerConstraintDTO)`.

Reply:
Ok